### DISCUSSION TITLE
Make SecretStr hashable

### DISCUSSION THREAD

**Discussion by chbndrhnns** [pydantic/pydantic#3499] [Feature Request]
Hi,

Is there a reason why the `SecretStr` type is not hashable? I use it in classes which subclass `BaseSettings` which I need to be hashable and thus I was wondering now about `SecretStr`.

Is there anything wrong with my approach here?

```py
class HashableSecretStr(SecretStr):
    """HashableSecretStr"""

    def __hash__(self):
        return hash(self.get_secret_value())
```

**Comment by NowanIlfideme** [comment]
Intuitively, it seems like this might not be secure, but given Python salts the `hash()` function I can't immediately say why it's bad.

**Reply by mssalvatore** [reply]
I can't think of any reason why this would pose a security risk.

**Reply by samuelcolvin** [reply]
maybe a typo?

Do you mean "I can think..." or "I can't think..."?

**Reply by mssalvatore** [reply]
Yeah, edited. Thanks.

**Comment by samuelcolvin** [comment]
I'd be happy to accept this, yes in theory you could use this to check if the internal value matched a value, but `SecretStr` is only designed to prevent mistaken printing, it's not a secure wrapper.

**Reply by mssalvatore** [reply]
Looks like this was already fixed in v1.10.0.

https://github.com/pydantic/pydantic/issues/4387

**Reply by samuelcolvin** [reply]
good point :facepalm:.

**Reply by chbndrhnns** [reply]
I guess I forgot to close the discussion after the PR was merged.

**Reply by samuelcolvin** [reply]
:+1:

Can't close discussions. :disappointed:
